app/services/nic_ip_required.rb

```diff
@@ -0,0 +1,8 @@
+module NicIpRequired
+  FIELDS = [:ip, :ip6].freeze
+
+  def self.new(field, *args)
+    raise ::Foreman::Exception.new(N_("Unknown address field - can't continue")) unless FIELDS.include?(field)
+    "NicIpRequired::#{field.to_s.camelize}".safe_constantize.new(*args)
```

Comment:
I think this magic is unnecessary - in both cases in Nic::Base you can just call `NicIpRequired::Ip6.new` or Ip4 as desired.